aws/s3/s3.go

```diff
@@ -78,6 +78,26 @@ func getConfig() (aws.Config, error) {
 	if os.Getenv("AWS_REGION") == "" {
 		return aws.Config{}, errors.New("AWS_REGION is not set")
 	}
+
```

Comment:
Can we declare `cfg` before L82, so we can do:
```
var cfg aws.Config
if awsEndpoint .... {
  cfg, err = config.LoadDefaultConfig(....)
} else {
  cfg, err = config.LoadDefaultConfig(....)
}

return cfg, err
```
So there will be only one return path.